### ISSUE TITLE
Users can still share public links to bookmarks folders even when 'Allow users to share via link' is disabled

### ISSUE THREAD

**Issue by NoOpFlop** [nextcloud/bookmarks#1358]
**Describe the bug**
When 'Allow users to share via link' is disabled, users can still create and share a link to their bookmarks folder. The link works even when the user is logged off. The link produced is in the format of "xxx.xxx.xxx.xxx/apps/bookmarks/public/(random string)"
It appears that the Bookmarks app may be ignoring Nextcloud's sharing permissions.

*In this scenario, only sharing the folders with other authenticated users, and only within their groups, is desired.*

Conditions:

1. Public sharing is off
2. Federated sharing all disabled
3. 'Allow resharing' is disabled
4. 'Default sharing' section is all unchecked
3. 'Restrict users to only share within their groups' is enabled
4. '+ Apps' > Bookmarks app > 'Limit to groups' is checked, and groups are selected.

**To Reproduce**

1. Go to bookmarks
2. Create a folder
3. Click '...' and select 'Details'
4. Select 'Sharing'
5. Click '+' next to 'Share link'
6. Share is created, link is in clipboard, and pasted link works externally without any form of security.

**Expected behavior**
'Share link' on folder shouldn't be there, or say 'disabled' - only 'select user or group' field should work.
(Additionally, I think that password protected (public) link sharing (when used) should be honored as well.)

**Desktop (please complete the following information):**
 - OS: Windows, Linux
 - Browser: Firefox, Chrome, Brave

**Server (please complete the following information):**
 - OS: Truenas (jailed instance of Nextcloud)
 - HTTP server: nginx 1.18.0_34,2
 - Database: mysql 5.7.32
 - PHP version: 7.4.13_2
 - Nextcloud version: 20
 - Bookmarks app version: 4.0.5

**Comment by marcelklehr** [comment]
Indeed, the app currently doesn't follow these settings. I used to think those only applied to the files app, though.

**Comment by electrofloat** [comment]
Any update on this one?

**Comment by marcelklehr** [comment]
Thank you for your patience, this will be part of the next release.
